packages/e2e-test-utils-playwright/src/request-utils/rest.ts

```diff
@@ -22,18 +23,31 @@ function splitRequestsToChunks( requests: BatchRequest[], chunkSize: number ) {
 }
 
 async function getAPIRootURL( request: APIRequestContext ) {
-	// Discover the API root url using link header.
-	// See https://developer.wordpress.org/rest-api/using-the-rest-api/discovery/#link-header
-	const response = await request.head( WP_BASE_URL );
-	const links = response.headers().link;
-	const restLink = links?.match( /<([^>]+)>; rel="https:\/\/api\.w\.org\/"/ );
-
-	if ( ! restLink ) {
-		throw new Error( `Failed to discover REST API endpoint.
- Link header: ${ links }` );
-	}
-
-	const [ , rootURL ] = restLink;
+	let restLink: unknown = null;
```

Comment:
Nit: Let's make it `RegExpMatchArray | null`.
```suggestion
	let restLink: RegExpMatchArray | null = null;
```

Reply:
The TS gymnastics here was that it needs to be casted as `unknown` first so that it can be casted as `RegExpMatchArray` later. TS does not work with expect assertions as it does with `if`s so if we cast it as you suggested, we'll need an additional `if ( ! restLink )` after the `expect.poll`, which is obsolete as the expect will throw if the `restLink` is `null`. Does that make sense or am I missing something?